lib/puppet/resource/catalog.rb

```diff
@@ -178,9 +178,9 @@ def apply(options = {})
         transaction.evaluate
       end
     rescue Puppet::Error => detail
-      Puppet.log_exception(detail, "Could not apply complete catalog: #{detail}")
+      raise Puppet::Error, ("Could not apply complete catalog: #{detail}"), detail.backtrace 
     rescue => detail
-      Puppet.log_exception(detail, "Got an uncaught exception of type #{detail.class}: #{detail}")
+      raise Puppet::Error, "Got an uncaught exception of type #{detail.class}: #{detail}", detail.backtrace
```

Comment:
Wouldn't it be cleaner to just let the original exception pass by? Converting to `Puppet::Error` with a generic message does not seem to add value.